sdk/src/types/block/error.rs

```diff
@@ -261,6 +270,21 @@ impl fmt::Display for Error {
                 "storage deposit return of {deposit} exceeds the original output amount of {amount}"
             ),
             Self::InvalidContextInputCount(count) => write!(f, "invalid context input count: {count}"),
+            Self::InvalidAddressWeight(w) => write!(f, "invalid address weight: {w}"),
+            Self::InvalidMultiAddressThreshold(t) => write!(f, "invalid multi address threshold: {t}"),
+            Self::InvalidMultiAddressCumulativeWeight {
+                cumulative_weight,
+                threshold,
+            } => {
+                write!(
+                    f,
+                    "invalid multi address cumulative weight {cumulative_weight} < threshold {threshold}"
+                )
+            }
+            Self::InvalidWeightedAddressCount(count) => write!(f, "invalid weighted address count: {count}"),
+            Self::WeightedAddressesNotUniqueSorted => {
```

Comment:
didnt we address sorting using a ordered list or something before?

Reply:
No, there was an attempt PR but it's been closed

Reply:
i see, treeset isnt boxed either, too bad